### ISSUE TITLE
ArithmeticException: divide by zero in EventManager

### ISSUE THREAD

**Issue by OscarSpruit** [urbanairship/android-library#147]
# Preliminary Info

### What Airship dependencies are you using?
com.urbanairship.android:urbanairship-fcm:12.2.0

### What are the versions of any relevant development tools you are using?
Not applicable.

# Report

### What unexpected behavior are you seeing?
Our app crashes on startup for some users because of an `ArithmeticException` in `com.urbanairship.analytics.data.EventManager`

### What is the expected behavior?
Obviously, it should not crash.

### What are the steps to reproduce the unexpected behavior?
I have no clue.

### Do you have logging for the issue?
```
Fatal Exception: java.lang.ArithmeticException: divide by zero
       at com.urbanairship.analytics.data.EventManager.addEvent(EventManager.java)
       at com.urbanairship.analytics.AnalyticsJobHandler.onUploadEvents(AnalyticsJobHandler.java)
       at com.urbanairship.analytics.AnalyticsJobHandler.onUploadEvents(AnalyticsJobHandler.java)
       at com.urbanairship.analytics.Analytics.access$600(Analytics.java)
       at com.urbanairship.job.Job$1.run(Job.java)
       at com.urbanairship.util.SerialExecutor$1.run(SerialExecutor.java)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
       at com.urbanairship.util.AirshipThreadFactory$1.run(AirshipThreadFactory.java)
       at java.lang.Thread.run(Thread.java:762)
```

### My additional input
Recently, we have changed the flow for dealing with user consent of third party analytics. On startup of our app we initialize `Airship` with analytics disabled and do a network request to fetch the users consent choices. Based on on the result of this request we enable/disable analytics for Airship again. My assumption would be that this crash could be caused by toggling the analytics too fast after Airship just got initialized. 

I dug around a bit in the Airship code to see why the exception occurs. I think `this.eventResolver.getDatabaseSize()` returns 0, which causes `avgSize` to be 0 and the second divsion to fail with an `ArithmeticException` (see code block below). I have no idea how my theory above could cause this, but I hope you can shine some more light on this 🙂 
```
// com.urbanairship.analytics.data.EventManager, lines 116-117
int avgSize = this.eventResolver.getDatabaseSize() / eventCount;
int batchEventCount = Math.min(500, this.preferenceDataStore.getInt("com.urbanairship.analytics.MAX_BATCH_SIZE", 512000) / avgSize);
```

**Comment by crow** [comment]
@OscarSpruit We'll look into this immediately - thanks for reporting.

**Comment by rlepinski** [comment]
The stack trace looks off to me. That should be uploadEvents not addEvent. Was it an out of date proguard mapping?

**Comment by OscarSpruit** [comment]
I found this odd as well, I guess it's indeed an old/incorrect proguard mapping 🤔

**Comment by rlepinski** [comment]
I think whats happening is database size is 0 and event count is > 0 probably due to a race condition of deleted the events in the middle of an upload task. Will try to fix this week.

**Comment by rlepinski** [comment]
Fixed in 12.2.3
